libs/ngx-dynamic-form-builder/src/lib/utils/dynamic-form-group.ts

```diff
@@ -406,115 +143,498 @@ export class DynamicFormGroup<TModel> extends FormGroup {
+
+                return getIsValidResult(isValid, validationMetadata);
+              };
+
+              fieldDefinition.validation.push(customValidation);
+
+              // Set field data
+              if (fields[fieldName] instanceof DynamicFormGroup) {
+                fields[fieldName].object = fields[fieldName].fields;
+              }
+
+              if (fieldDefinition.data === undefined) {
+                fieldDefinition.data = fields[fieldName];
+              }
+            }
+
+            // Handle remaining validation
+            else {
+
+              // create validation function and add to field validation
+              const dynamicValidate = function (control: FormControl) {
+
+                if (!control) {
+                  return null;
+                }
+
+                let isValid = (control.parent && control.parent.value)
+                  ? validator.validateValueByMetadata(control.value, validationMetadata)
+                  : true;
+
+                if (!isValid && conditionalValidations.length > 0) {
+                  const validateErrors = setObjectValueAndGetValidationErrors(control, fieldName, validatorOptions);
+                  isValid = validateErrors.filter((error: ValidationError) => error.property === fieldName).length === 0;
                 }
+
+                return getIsValidResult(isValid, validationMetadata);
+              };
+
+              fieldDefinition.validation.push(dynamicValidate);
+
+              // Set field data
+              if (fields[fieldName] instanceof DynamicFormGroup) {
+                fields[fieldName].object = fields[fieldName].fields;
+              }
+
+              if (fieldDefinition.data === undefined) {
+                fieldDefinition.data = fields[fieldName];
               }
+
             }
           }
         }
       }
     });
+
+    // Convert to a structure, angular understands
+    if (fieldDefinition.data instanceof DynamicFormGroup || fieldDefinition.data instanceof FormArray) {
+      formGroupFields[fieldName] = fieldDefinition.data;
+    }
+    else {
+      formGroupFields[fieldName] = [fieldDefinition.data, fieldDefinition.validation];
+    }
+  });
+
+  return formGroupFields;
+}
+
+// ******************************************************************************************
+// Helper functions to help make the main code more readable
+//
+
+function isPropertyValidatorOfType(validationMetadata: ValidationMetadata, fieldName: string, validationMetadataType: string) {
+  return validationMetadata.propertyName === fieldName && validationMetadata.type === validationMetadataType;
+}
+
+function hasValidationRule(validationMetadata: ValidationMetadata, validator: Validator, typeKey: string) {
+  return validationMetadata.type === ValidationTypes[typeKey] && validator[validationMetadata.type] === undefined;
+}
+
+function getAllErrors(validateErrors: ValidationError[], key: string): ValidationError[] {
```

Comment:
Moved out to a function to help with readability of the core logical code. I have a question, per the note below
```
// NOTE: Not exactly sure what this is supposed to do. It looks like it will always return true if the array.length > 0.
// Could substitute the function with a length check?
```